src/mapper/pkg/resolvers/schema.resolvers.go

```diff
@@ -21,40 +21,63 @@ import (
 	"golang.org/x/exp/slices"
 )
 
-func (r *mutationResolver) ResetCapture(ctx context.Context) (bool, error) {
+func (r *mutationResolver) ResetCapture(_ context.Context) (bool, error) {
 	logrus.Info("Resetting stored intents")
 	r.intentsHolder.Reset()
 	return true, nil
 }
 
-func (r *mutationResolver) ReportCaptureResults(ctx context.Context, results model.CaptureResults) (bool, error) {
-	for _, captureItem := range results.Results {
-		srcPod, err := r.kubeFinder.ResolveIpToPod(ctx, captureItem.SrcIP)
-		if err != nil {
-			if errors.Is(err, kubefinder.ErrFoundMoreThanOnePod) {
-				logrus.WithError(err).Debugf("Ip %s belongs to more than one pod, ignoring", captureItem.SrcIP)
-			} else {
-				logrus.WithError(err).Debugf("Could not resolve %s to pod", captureItem.SrcIP)
-			}
-			continue
+func (r *mutationResolver) discoverSrcIdentity(ctx context.Context, src model.RecordedDestinationsForSrc) *model.OtterizeServiceIdentity {
+	srcPod, err := r.kubeFinder.ResolveIpToPod(ctx, src.SrcIP)
+	if err != nil {
+		if errors.Is(err, kubefinder.ErrFoundMoreThanOnePod) {
+			logrus.WithError(err).Debugf("Ip %s belongs to more than one pod, ignoring", src.SrcIP)
+		} else {
+			logrus.WithError(err).Debugf("Could not resolve %s to pod", src.SrcIP)
 		}
+		return nil
+	}
+	if src.SrcHostname != "" && srcPod.Name != src.SrcHostname {
+		// This could mean a new pod is reusing the same IP
+		// TODO: Use the captured hostname to actually find the relevant pod (instead of the IP that might no longer exist or be reused)
+		logrus.Warnf("Found pod %s (by ip %s) doesn't match captured hostname %s, ignoring", srcPod.Name, src.SrcIP, src.SrcHostname)
+		return nil
+	}
 
-		if srcPod.DeletionTimestamp != nil {
-			logrus.Debugf("Pod %s is being deleted, ignoring", srcPod.Name)
+	if srcPod.DeletionTimestamp != nil {
+		logrus.Debugf("Pod %s is being deleted, ignoring", srcPod.Name)
+		return nil
+	}
+
+	srcService, err := r.serviceIdResolver.ResolvePodToServiceIdentity(ctx, srcPod)
+	if err != nil {
+		logrus.WithError(err).Debugf("Could not resolve pod %s to identity", srcPod.Name)
+		return nil
+	}
+	filteredDestinations := make([]model.Destination, 0)
+	for _, dest := range src.Destinations {
+		if srcPod.CreationTimestamp.After(dest.LastSeen) {
+			logrus.Debugf("Pod %s was created after capture time %s, ignoring", srcPod.Name, dest.LastSeen)
 			continue
 		}
+		filteredDestinations = append(filteredDestinations, dest)
+	}
```

Comment:
This one is actually a bug since this array is unused, so there is no actual filtering of packets older than pod creation timestamp